test/api.test.ts

```diff
@@ -21,10 +21,42 @@ suite(`AzureFunctionsExtensionApi`, () => {
         api = nonNullValue(extension).exports.getApi<AzureFunctionsExtensionApi>('^1.0.0');
     });
 
+    test('createFunction in a subfolder of a workspace', async () => {
+        const functionName: string = 'endpoint1';
+        const language: string = ProjectLanguage.JavaScript;
+        const workspaceFolder = getTestWorkspaceFolder();
+        const relativeProjectFolder = 'api';
+        const folderPath: string = path.join(workspaceFolder, relativeProjectFolder);
+
+        await runWithInputs('api.createFunction', [language, functionName], registerOnActionStartHandler, async () => {
+            await api.createFunction({
+                folderPath,
+                suppressCreateProjectPrompt: true,
+                suppressOpenFolder: true,
+                templateId: 'HttpTrigger',
+                languageFilter: /Python|C\#|^(Java|Type)Script$/i,
+                functionSettings: { authLevel: 'anonymous' },
+                targetFramework: 'netcoreapp3.1' // Will only work on functions api v1.4.0, but won't hurt on v1.3.0
+            });
+        });
+
+        const validateOptions: IValidateProjectOptions = getJavaScriptValidateOptions(true, undefined, relativeProjectFolder);
+        validateOptions.expectedPaths.push(
+            path.join(relativeProjectFolder, functionName, 'index.js'),
+            path.join(relativeProjectFolder, functionName, 'function.json'),
+            path.join(relativeProjectFolder, 'package.json')
+        );
+        validateOptions.expectedSettings
+        // Exclude .git because the test workspace folders are already inside a git repo so we don't do git init.
+        validateOptions.excludedPaths?.push('.git');
+        validateOptions.workspaceFolder = workspaceFolder;
```

Comment:
I think I still want to still add `workspaceFolder` to the options, and not just `projectSubpath`. But I can do that in the `getJavaScriptValidateOptions` if you want.

I need the `workspacePath` in `validateProject` because even though I could, I'm not sure we want to derive the workspace path from "subtracting" the `projectSubpath` from the `projectPath`. Or is that what you wanted to do?

Reply:
It's not worth going back and forth on - I already approved, so feel free to do what you think is best